tools/pythonpkg/src/pyconnection.cpp

```diff
@@ -969,7 +969,13 @@ unique_ptr<DuckDBPyRelation> DuckDBPyConnection::Table(const string &tname) {
 	if (qualified_name.schema.empty()) {
 		qualified_name.schema = DEFAULT_SCHEMA;
 	}
-	return make_uniq<DuckDBPyRelation>(connection->Table(qualified_name.schema, qualified_name.name));
+	try {
+		return make_uniq<DuckDBPyRelation>(connection->Table(qualified_name.schema, qualified_name.name));
+	} catch (const CatalogException &e) {
+		// CatalogException will be of the type '... is not a table'
+		// Not a table in the database, make a query relation that can perform replacement scans
+		return RunQuery(StringUtil::Format("from %s", tname), tname);
```

Comment:
This should use `KeywordHelper::WriteOptionallyQuoted`, otherwise this will fail on relations that contain e.g. spaces, are keywords, etc. Perhaps you could add a test with a dataframe called e.g. "select":

```py
select = pd.DataFrame.from_dict(..)
duckdb.table('select')
```

Reply:
Another option is to construct the `SelectStatement` programmatically instead of forwarding this to the parser, but this is more work